lib/libadmin/util.c

```diff
@@ -142,7 +150,15 @@ cookieValue(char *var, char *val)
             cookie = STRDUP(cookie);
             numVars = 0;
             vars = (char **)MALLOC(sizeof(char *));
+            if (vars == NULL) {
+                return NULL;
+            }
             vals = (char **)MALLOC(sizeof(char *));
+            if (vals == NULL) {
+                FREE(vars);
+                return NULL;
```

Comment:
Should we also free `cookie` before the returns?

Reply:
Good catch I missed the STRDUP.
We should also check if STRDUP fails ....